cmd/acme/cert.go

```diff
@@ -215,11 +220,27 @@ func pollCert(url string) [][]byte {
 	}
 }
 
-func challengeFile(domain, content string) (string, error) {
-	f, err := ioutil.TempFile("", domain)
+func challengeFile(challengeFilename, domain, content string) (string, error) {
+	// If ACME_CHALLENGE_DIR is set, place the challenge into it.
+	// Otherwise, create a temporary file
+	var f *os.File
+	var err error
+	if os.Getenv("ACME_CHALLENGE_DIR") != "" {
+		// Create .well-known directory
+		challengeDir := filepath.Join(os.Getenv("ACME_CHALLENGE_DIR"), ".well-known/acme-challenge")
```

Comment:
I mean to append it automatically unless already present in the path. But what you're saying makes sense, I agree. It's important this is documented, e.g. "challenge dir should point to the root of your website".